### PR TITLE
[5.1] Fix for Query Builder not clearing the select bindings before an aggregate.

### PR DESCRIPTION
**Pull request by dtaub**
Currently, before running an aggregate function, the Query Builder clears and resets the columns, but not the select bindings. This is an issue when the query contains a sub-select query with bindings, since the aggregate query will have extraneous bindings.

The use case that caused me to run into this was pagination, which uses count() internally.

### CODE REVIEW

File: src/Illuminate/Database/Query/Builder.php [hunk 1]
```diff
@@ -1641,7 +1641,9 @@ public function aggregate($function, $columns = ['*'])
         $this->aggregate = compact('function', 'columns');
 
         $previousColumns = $this->columns;
-
+        $previousSelectBindings = $this->bindings['select'];
+        $this->bindings['select'] = [];
+
```
Comment (GrahamCampbell) [inline src/Illuminate/Database/Query/Builder.php]: cs

**Comment by dtaub** [comment]
@GrahamCampbell, by CS are you referring to the whitespace on 1646? 

Also, is there any guide on checking the coding style for Laravel? php-cs-fixer is telling me all the docblocks are formatted wrong.

**Comment by GrahamCampbell** [comment]
> Also, is there any guide on checking the coding style for Laravel? php-cs-fixer is telling me all the docblocks are formatted wrong (even if I pass in the .php_cs config file).

Yes, our php_cs fixer file documents this. I don't know what's going on if you can't get it to work. Works fine dor me.

**Comment by dtaub-mms** [comment]
Turns out I had a typo in the path to the .php_cs file...

I fixed the cs issue and also added an assertion to make sure the bindings are restored after the aggregate.

**Comment by dtaub** [comment]
Turns out I had a typo in the path to the .php_cs file...

I fixed the cs issue and also added an assertion to make sure the bindings are restored after the aggregate.

**Comment by magnosis** [comment]
I got here from #14492 (closed) and I am running into the same problem.

$query->withCount('foo')->having('foo_count', '<', \DB::raw('bar'))->get() works.
$query->withCount('foo')->having('foo_count', '<', \DB::raw('bar'))->exists() works.
$query->withCount('foo')->having('foo_count', '<', \DB::raw('bar'))->count() throws exception.

What is the fix for that last one?

- Laravel Version: 5.7.28
- PHP Version: 7.1.23
- Database Driver & Version: MySQL 14.14

**Comment by GrahamCampbell** [comment]
Please open a new issue rather than commenting on a very old PR. :)

**Pull request merged**